Importer/Log/CsvLog.php

```diff
@@ -50,9 +50,12 @@ public function addDataError($index, $errorMessage)
     /**
      * {@inheritdoc}
      */
-    public function addCriticalError($errorMessage)
+    public function addCriticalError($message, $userError = false)
```

Comment:
The idea was to have an `errorReason` as "lower level", internal description of an error, e.g. "DBAL exception, foreign key constraint violation", and `userMessage` as "user-land", user interface friendly description, e.g. "Referenced object does not exist".
The `errorReason` is required, because this is actually a critical error message (that can be put to application logs), and `userMessage` is optional, because sometimes it doesn't make sense to show it to user, simple "An application error occurred" is fine.

For "Tuesday" example, assuming that you really want to inform user about it, `errorReason` might contain technical details, e.g. context, and `userMessage` is just message for the user ("We don't work on Tuesdays, try tomorrow").